superset/sql_parse.py

```diff
@@ -432,6 +519,192 @@ def get_settings(self) -> dict[str, str]:
         }
 
 
+class KQLSplitState(enum.Enum):
+    """
+    State machine for splitting a KQL query.
+
+    The state machine keeps track of whether we're inside a string or not, so we
+    don't split the query in a semi-colon that's part of a string.
+    """
+
+    OUTSIDE_STRING = enum.auto()
+    INSIDE_SINGLE_QUOTED_STRING = enum.auto()
+    INSIDE_DOUBLE_QUOTED_STRING = enum.auto()
+    INSIDE_MULTILINE_STRING = enum.auto()
+
+
+def split_kql(kql: str) -> list[str]:
```

Comment:
I don't see a test for this

Reply:
It's tested indirectly here: 

https://github.com/apache/superset/pull/27522/files#diff-7e7fc3bcddf707b5ecdbd5b2f0968f957f49fe14fe082dafce51204726940081R1901

But I'll add some explicit tests.